src/daft-scan/src/glob.rs

```diff
@@ -0,0 +1,154 @@
+use std::{fmt::Display, sync::Arc};
+
+use common_error::DaftResult;
+use daft_core::schema::SchemaRef;
+use daft_io::{get_io_client, get_runtime, IOStatsContext};
+
+use crate::{DataFileSource, FileType, PartitionField, ScanOperator, ScanOperatorRef, ScanTask};
+#[derive(Debug)]
+pub struct GlobScanOperator {
+    glob_path: String,
+    file_type: FileType,
+    columns_to_select: Option<Vec<String>>,
+    limit: Option<usize>,
+    schema: SchemaRef,
+    io_config: Arc<daft_io::IOConfig>,
+}
+
+fn run_glob(
+    glob_path: &str,
+    io_config: Arc<daft_io::IOConfig>,
+    limit: Option<usize>,
+) -> DaftResult<Vec<String>> {
+    // Use single-threaded runtime which should be safe here as it is not shared across async contexts
+    let runtime = get_runtime(false)?;
+    let io_client = get_io_client(false, io_config)?;
```

Comment:
I'm actually not super sure if this is safe, since this may grab a different "current-thread" runtime every time, but `get_io_client(false, io_config)` will retrieve the same IOClient for a given thread.

Reply:
Using multithreaded runtime instead after offline discussion